packages/compass-workspace/src/components/workspace-tabs.spec.tsx

```diff
@@ -0,0 +1,244 @@
+import React from 'react';
+import { render, screen, cleanup } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { expect } from 'chai';
+import sinon from 'sinon';
+
+import { WorkspaceTabs, getTabType } from './workspace-tabs';
+
+describe('WorkspaceTabs', function () {
+  let onCreateNewTabSpy: sinon.SinonSpy;
+  let onCloseTabSpy: sinon.SinonSpy;
+  let onSelectSpy: sinon.SinonSpy;
+  let onMoveTabSpy: sinon.SinonSpy;
+
+  beforeEach(function () {
+    onCreateNewTabSpy = sinon.spy();
+    onCloseTabSpy = sinon.spy();
+    onSelectSpy = sinon.spy();
+    onMoveTabSpy = sinon.spy();
+  });
+
+  afterEach(cleanup);
+
+  describe('when rendered', function () {
+    beforeEach(function () {
+      render(
+        <WorkspaceTabs
+          onCreateNewTab={onCreateNewTabSpy}
+          onCloseTab={onCloseTabSpy}
+          onSelectTab={onSelectSpy}
+          onMoveTab={onMoveTabSpy}
+          tabs={[]}
+        />
+      );
+    });
+
+    it('should render a create new tab button', async function () {
+      expect(await screen.findByLabelText('Create new tab')).to.be.visible;
+    });
+
+    it('should call to create a new tab when the create new tab button is clicked', async function () {
+      expect(onCreateNewTabSpy.callCount).to.equal(0);
+
+      const newTabButton = await screen.findByLabelText('Create new tab');
+      newTabButton.click();
+
+      expect(onCreateNewTabSpy.callCount).to.equal(1);
+    });
+  });
+
+  describe('when rendered with multiple tabs', function () {
+    beforeEach(function () {
+      render(
+        <WorkspaceTabs
+          onCreateNewTab={onCreateNewTabSpy}
+          onCloseTab={onCloseTabSpy}
+          onSelectTab={onSelectSpy}
+          onMoveTab={onMoveTabSpy}
+          tabs={[
+            {
+              namespace: 'mock-tab-1',
+              id: '1',
+              activeSubTabName: 'Documents',
+              isActive: false,
+              isTimeSeries: false,
+              isReadonly: false,
+            },
+            {
+              namespace: 'mock-tab-2',
+              id: '2',
+              activeSubTabName: 'Schema',
+              isActive: true,
+              isTimeSeries: true,
+              isReadonly: false,
+            },
+            {
+              namespace: 'mock-tab-3',
+              id: '3',
+              activeSubTabName: 'Aggregation',
+              isActive: false,
+              isTimeSeries: false,
+              isReadonly: true,
+            },
+          ]}
+        />
+      );
+    });
+
+    it('should render all of the tab namespaces', function () {
+      expect(screen.getByText('mock-tab-1')).to.be.visible;
+      expect(screen.getByText('mock-tab-2')).to.be.visible;
+      expect(screen.getByText('mock-tab-3')).to.be.visible;
+    });
+
+    it('should render all of the tab subtab names', function () {
```

Comment:
Why do we want this component to do this?

Reply:
We want the tabs to contains things like `Document` or `Explain Plan` that indicate the contents of the tab.
This test checks that the tabs rendered by `workspace-tabs` contain the correct information passed through. It's definitely a bit of a coupled test as it pairs the rendering in `tab` with `workspace-tabs`, but I think that helps build some guarantee that things are rendered and passed through. Does that answer the question?

Reply:
Would it be better to rename this `activeSubTabName` to `workspaceName` to be less collection/subtab centric in the naming?

Reply:
Renaming it would definitely be better yeah. But I'll never be fond of this coupling here, in my opinion, `WorkspaceTabs` shouldn't know anything about sub-tabs, tabs content, and the fact that there is a concept of time-series and whatnot.

Reply:
Updated 👍  thanks for raising.